packages/components/src/templates/next/layouts/Collection/utils.ts

```diff
@@ -38,13 +52,22 @@ export const getAvailableFilters = (
       })),
     },
     {
-      id: "year",
+      id: FILTER_ID_YEAR,
       label: "Year",
-      items: Object.entries(years).map(([label, count]) => ({
-        id: label.toLowerCase(),
-        label,
-        count,
-      })),
+      items:
+        // do not show "not specified" option if all items have undefined dates
+        yearFilterItems.length === 0
+          ? []
+          : numberOfUndefinedDates === 0
+            ? yearFilterItems
+            : [
+                ...yearFilterItems,
```

Comment:
nit: same here regarding the ordering of the years, should be in ascending order

Reply:
okay!

anyway, will put this in backlog since its unrelated to this PR